File: cpp/iterador.cpp
#include <iostream>
#include <vector>
using namespace std;

class Iterador {
private:
    vector<string> array;
    int cursor = 0;

public:
    Iterador(const vector<string>& arr) : array(arr) {}

    bool has_next() {
        return cursor < array.size();
    }

    string next() {
        return array[cursor++];
    }
};

int main() {
    vector<string> dias = {"lunes", "martes", "miércoles", "jueves", "viernes"};
    Iterador it(dias);

    while (it.has_next()) {
        cout << "Todavía queda un elemento: " << it.next() << endl;
    }

    cout << "Ya he terminado" << endl;
    return 0;
}

File: java/Iterador.java
class IteradorBase {
    private String[] array;
    private int cursor = 0;

    public IteradorBase(String[] array) {
        this.array = array;
    }

    public boolean has_next() {
        return cursor < array.length;
    }

    public String next() {
        return array[cursor++];
    }
}

public class Iterador {
    public static void main(String[] args) {
        String[] dias = {"lunes", "martes", "miércoles", "jueves", "viernes"};
        IteradorBase it = new IteradorBase(dias);

        while (it.has_next()) {
            System.out.println("Todavía queda un elemento: " + it.next());
        }

        System.out.println("Ya he terminado");
    }
}
